expcore/Mini_games.lua

```diff
@@ -199,75 +211,77 @@ function Mini_games.error_in_game(error_game)
     game.print("an error has occured things may be broken, error: "..error_game)
 end
 local mini_game_list
+--gui
 local on_vote_click = function (player,element,event)
-    --local frame = Gui.get_left_element(player,mini_game_list).container
-    --local scroll = frame.scroll.table
-    local caption = tonumber(element.parent.parent["edit-1"]["race"].caption)+1
-    element.parent.parent["edit-1"]["race"].caption = tostring(caption)
+    local name = element.parent.name
+    local scroll_table = element.parent.parent
+    local mini_game = Mini_games.mini_games[name]
+    local args
+    if mini_game.gui_callback then
+        args = mini_game.gui_callback(scroll_table)
+    end
+    --main_gui[1] = element.parent.parent.parent.parent.parent.parent
+    for i,player in ipairs(game.connected_players) do
+        main_gui[i] = Gui.get_left_element(player,mini_game_list)
+        Gui.toggle_left_element(player,main_gui[1],false)
+    end
+    if args then
+        Mini_games.start_game(name,args)
+    else 
+        Mini_games.start_game(name)
+    end
+    Gui.update_top_flow(player) 
 end
 
 
+
 local vote_button =
 Gui.element{
     type = 'sprite-button',
     sprite = 'utility/check_mark',
-    style = 'quick_bar_slot_button',   
+    style = 'quick_bar_slot_button',
 }
 :on_click(on_vote_click)
 
 
-local amount_label = 
-Gui.element(function(_,parent)
-parent.add{
-    type = "label",
-    caption = '0',  
-    style ="heading_1_label",
-    name = "race"
-}
-    
-end)
-
 local add_mini_game =
-Gui.element(function(_,parent)
-    local vote_flow = parent.add{ type = 'flow', }
+Gui.element(function(_,parent,name)
+    local vote_flow = parent.add{ type = 'flow', name = name }
     vote_flow.style.padding = 0
-    vote_button(vote_flow)
+    vote_button(vote_flow,name)
```

Comment:
If vote button is used `Gui.element{}` syntax then why are you passing name? All arguments after the first are ignored unless you are using a custom draw function or a custom style function.

Reply:
Just forgot to remove that the button was a func at first.